e2e/testcases/helm_sync_test.go

```diff
@@ -47,6 +53,29 @@ const (
 
 var privateARHelmRegistry = fmt.Sprintf("oci://us-docker.pkg.dev/%s/config-sync-test-ar-helm", nomostesting.GCPProjectIDFromEnv)
 
+// deleteServiceLoadBalancerFirewall deletes the Firewall rule in GCP created
+// by cloud-provider-gce when a Service of type LoadBalancer is created.
+//
+// This is suppossed to be done by finalizer when the Service is deleted, but
+// it doesn't seem to be working.
+// https://github.com/kubernetes/cloud-provider-gcp/blob/master/providers/gce/gce_loadbalancer_internal.go#L325
+func deleteServiceLoadBalancerFirewall(nt *nomostest.NT, svc *corev1.Service) error {
+	loadBalancerName := cloudprovider.DefaultLoadBalancerName(svc)
+	fwName := gce.MakeFirewallName(loadBalancerName)
+	args := []string{
+		"gcloud", "compute", "firewall-rules", "delete", fwName,
+		"--project", nomostesting.GCPProjectIDFromEnv,
+	}
+	nt.T.Log(strings.Join(args, " "))
+	cmd := exec.Command(args[0], args[1:]...)
+	out, err := cmd.CombinedOutput()
+	if err != nil {
+		nt.T.Log(out)
+		return err
+	}
+	return nil
+}
+
 // TestPublicHelm can run on both Kind and GKE clusters.
 // It tests Config Sync can pull from public Helm repo without any authentication.
 func TestPublicHelm(t *testing.T) {
```

Comment:
`TestHydrateHelmComponents` also uses the remote ingress-nginx helm chart for testing: http://go/cs-oss/blob/main/e2e/testdata/hydration/helm-components/kustomization.yaml#L19.

We will need to add the cleanup in that test as well.

An alternative to bypass the cloud-provider-gce issue is to use a simple helm chart without ingress.

Reply:
I also would prefer to use a simpler helm chart

Reply:
Agreed. Then we don't need to worry about removing the code snippet when the cloud-provider-gce issue is fixed in the future.

Reply:
If y’all want to make the PR to a different helm chart, be my guest. I’m just trying to unblock CI. 

This is looking like a legitimate GKE bug. I’m following up on internal channels.

Reply:
Sure. I can make a PR that changes the ingress chart to wordpress.

Reply:
It’s being used in a half dozen places, including the nomos hydrate tests and helm tests. I’m not sure if those all need to be updated or not, because I haven’t narrowed down which Services are leaving GCP resources behind.

Reply:
https://github.com/GoogleContainerTools/kpt-config-sync/pull/372